crates/metrics/src/lib.rs

```diff
@@ -2,40 +2,131 @@
 
 use serde_derive::{Deserialize, Serialize};
 use std::{collections::HashMap, fs, io, path::Path, time::Duration};
+use sysinfo::{CpuExt, System, SystemExt};
 use thiserror::Error;
 
-/// Cycle-count metrics for a particular workload.
-///
-/// Stores the total cycle count and a breakdown of cycle count per named region.
-#[derive(Debug, Serialize, Deserialize, PartialEq, Eq)]
-pub enum WorkloadMetrics {
-    /// Metrics produced when benchmarking in execution mode
-    Execution {
-        /// Name of the workload (e.g., "fft", "aes").
-        name: String,
+/// Represents a single benchmark run.
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
+pub struct BenchmarkRun {
+    /// Name of the benchmark.
+    pub name: String,
+    /// Block used gas
+    pub block_used_gas: u64,
+    /// Execution metrics for the benchmark run.
+    #[serde(skip_serializing_if = "Option::is_none")]
```

Comment:
Something to note here is that: `skip_serializing_if` doesn't play well with some encoders like bincode and rkyv when you want to derserialize the structure and the field was None.

This is okay for json, so I think its not a problem